### PR TITLE
feat: steamid cond

### PR DESCRIPTION
**Pull request by tricksurf**
Added 'steamid=' condition that checks for specific partner Steam32 ID's. Allowing different commands for different partners.

### CODE REVIEW

**Comment by JaioCG** [comment]
Useful for having different demos folder for different partners, among other things

File: src/OffsetsData.hpp [hunk 1]
```diff
@@ -23,12 +23,13 @@ OFFSET_DEFAULT(GetSaveDirName, 124, 124)
 OFFSET_DEFAULT(ExecuteClientCmd, 104, 104)
 OFFSET_DEFAULT(GetActiveSplitScreenPlayerSlot, 127, 127)
 OFFSET_DEFAULT(GetSteamAPIContext, 177, 178)
-OFFSET_DEFAULT(GetMouseDelta, -1, 161) // This method only exists on Linux
+OFFSET_DEFAULT(GetMouseDelta, -1, 161)  // This method only exists on Linux
```
Comment (NeKzor) [inline src/OffsetsData.hpp:26]: This white-space change should be reverted.

File: src/SourceAutoRecord.vcxproj [hunk 2]
```diff
@@ -14,7 +14,7 @@
     <ProjectGuid>{3F6459A9-566E-4CAC-A412-C03FF51D67E9}</ProjectGuid>
     <RootNamespace>SourceAutoRecord</RootNamespace>
     <ProjectName>SourceAutoRecord</ProjectName>
-    <WindowsTargetPlatformVersion>10.0.18362.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
```
Comment (NeKzor) [inline src/SourceAutoRecord.vcxproj:17]: This change should not be committed.

File: src/Features/ConfigPlus.cpp [hunk 3]
```diff
@@ -377,6 +378,7 @@ static void FreeCondition(Condition *c) {
 		FreeCondition(c->binop_l);
 		FreeCondition(c->binop_r);
 		break;
+	case Condition::STEAMID:
```
Comment (NeKzor) [inline src/Features/ConfigPlus.cpp:381]: This causes a memory leak. You want to free `c->val` like in the other conditions.

**Review by NeKzor — CHANGES_REQUESTED** [review]

File: src/Features/ConfigPlus.cpp (hunk 3 above)
Comment (ThisAMJ) [reply src/Features/ConfigPlus.cpp:381]: I would've put this up with the other conditions that use the same mechanism, but this works

**Pull request closed without merging**